src/reconstruct/helper.ts

```diff
@@ -0,0 +1,143 @@
+import { NodeId, ParentInformation, RNode, RType } from '../r-bridge'
+import { SourcePosition } from '../util/range'
+import { ReconstructionConfiguration } from './reconstruct'
+
+
+/*
+--helper function--
+*/
+export type Selection = Set<NodeId>
+interface PrettyPrintLinePart {
+	part: string
+	loc:  SourcePosition
+}
+export interface PrettyPrintLine {
+	linePart: PrettyPrintLinePart[]
+	indent:   number
+}
+
+/**
+ * Splits text on '\n' to create lineParts and encapsulates them in the Code type
+ */
+export function plain(text: string, location: SourcePosition): Code {
+	let part = ''
+	const printLine: PrettyPrintLine = {linePart: [], indent: 0}
+	for(let character = 0; character < text.length; character++) {
+		const element = text[character]
+		if(element === '\n') {
+			printLine.linePart.concat({part: part, loc: location})
+		}
+		else {
+			part = part.concat(element)
+		}
+	}
+	return [printLine]
+}
+
+/**
+ * this function will merge up to n code pieces into a singular code piece, garanting that there are no duplicate lines and all lines are in order
+ */
+export function merge(snipbits: Code[]): Code {
+	const buckets: PrettyPrintLine[] = []
+	const result:Code = []
+
+	//seperate and group lineParts by lines
+	snipbits.forEach(code => {
```

Comment:
Counterintuitively (and depending on the js/node setup) this can be significantly slower than your off-the-shelf `for-of` loop. So it may be better to switch for this hotter piece